tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2725,16 +2725,17 @@ class NoImportTypingFromTypeCheck(BaseChecker):
 
     def visit_importfrom(self, node):
         """Check that we aren't importing from typing under if TYPE_CHECKING."""
-        if isinstance(node.parent, astroid.If) and (node.modname == "typing" or node.modname == "typing_extensions"):
-            self.add_message(
-                msgid=f"no-typing-import-in-type-check",
-                node=node,
-                confidence=None,
-            )
+        if isinstance(node.parent, astroid.If) and not node.parent.has_elif_block() and node.parent.orelse == []:
+            if node.modname == "typing" or node.modname == "typing_extensions":
+                self.add_message(
+                    msgid=f"no-typing-import-in-type-check",
+                    node=node,
+                    confidence=None,
+                )
 
     def visit_import(self, node):
         """Check that we aren't importing from typing under if TYPE_CHECKING."""
-        if isinstance(node.parent, astroid.If):
+        if isinstance(node.parent, astroid.If) and not node.parent.has_elif_block() and node.parent.orelse == []:
```

Comment:
ensures it is an if on its own that we are checking, there is no elif or else contents